typescript-json-schema.ts

```diff
@@ -859,7 +862,7 @@ export class JsonSchemaGenerator {
             // We don't return the full definition, but we put it into
             // reffedDefinitions below.
             returnedDefinition = {
-                $ref:  "#/definitions/" + fullTypeName
+                $ref:  `${this.args.id}#/definitions/` + fullTypeName
```

Comment:
Can you show me an example of this used somewhere?

Reply:
if I understand your question correct, it's how ajv expects to work with refs https://github.com/epoberezkin/ajv#ref

Reply:
Is this in the JSON Schema standard?

Reply:
I think so) http://json-schema.org/latest/json-schema-core.html#rfc.section.8.2

Reply:
But this is to access things from another schema, right? Why would I want to add an id to refer to stuff within the same file?

Reply:
It's a good question for owner of that library. From what I saw they relay on this id as key in object of multiple schemas. And without such `$id` property they just set schema to empty string key, and it doesn't work properly.

I can describe whole use case where it broken. When I specify file path pattern through --include, it puts all my actual interfaces as definitions, so I don't have some root schema and only a bunch of definitions.

for example:
``` typescript
// file interfaces/Interface1.root.ts
import SubInterface1 from '../shared/SubInterface1';
interface Interface1 {
  prop1: SubInterface1;
}

// file interfaces/Interface2.root.ts
import SubInterface1 from '../shared/SubInterface1';
interface Interface2 {
  prop1: SubInterface1
}

// file shared/SubInterface1.ts
interface Interface1 {
  prop1: SubInterface1
}
```

after that with command like `typescript-json-schema tsconfig.json "*" --include=interfaces/*.ts`

we will get a schema of definitions without any root object.

after that I set my schema for `ajv` like
``` typescript
import schema from 'schema.ts';
ajv.addSchema(schema);
...
```

and when I want to validate some specific data I put concrete part of this file:
```typescript
...
ajv.validate(schema.definitions.Interface1, objectToValidate1)
ajv.validate(schema.definitions.Interface2, objectToValidate2)
```